controllers/openstackmachine_controller.go

```diff
@@ -353,10 +353,17 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be got or created: %v", err))
 			return ctrl.Result{}, nil
 		}
+
 		err = computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP)
 		if err != nil {
-			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be associated: %v", err))
-			return ctrl.Result{}, nil
+			if apierrors.IsNotFound(err) {
+				time.Sleep(2 * time.Second)
```

Comment:
What Do you think about something like this? (untested)

```go
		backoff := wait.Backoff{
			Steps:    10,
			Duration: time.Second,
			Factor:   1.25,
			Jitter:   0.1,
		}
		err = wait.ExponentialBackoff(backoff, func() (bool, error) {
			if err := computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP); err != nil {
				return false, nil
			}
			return true, nil
		})
		if err != nil {
			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be associated: %v", err))
			return ctrl.Result{}, nil
		}
```

Reply:
yes, this seems better approach, thanks~